src/box/module_cache.c

```diff
@@ -296,14 +296,15 @@ module_new(const char *package, size_t package_len,
 
 	memcpy(m->package, package, package_len);
 	m->package[package_len] = 0;
-
-	const char *tmpdir = getenv("TMPDIR");
-	if (tmpdir == NULL)
-		tmpdir = "/tmp";
+	const char *tmpdir = getenv_safe("TMPDIR", NULL, 0);
+	const char *print_dir = tmpdir;
+	if (print_dir == NULL)
+		print_dir = "/tmp";
 
 	char dir_name[PATH_MAX];
 	int rc = snprintf(dir_name, sizeof(dir_name),
-			  "%s/tntXXXXXX", tmpdir);
+			  "%s/tntXXXXXX", print_dir);
+	free((char *)tmpdir);
```

Comment:
Nit: Better define `tmpdir` as `char *` (without `const`) - it'll work as `getenv_safe` returns a non-const pointer.

Reply:
Sure. I changed `print_dir` to `char *` for consistency. No point to make it `const char *` anymore.
```diff
diff --git a/src/box/module_cache.c b/src/box/module_cache.c
index 9386ff625..3d9b1883d 100644
--- a/src/box/module_cache.c
+++ b/src/box/module_cache.c
@@ -296,15 +296,15 @@ module_new(const char *package, size_t package_len,
 
        memcpy(m->package, package, package_len);
        m->package[package_len] = 0;
-       const char *tmpdir = getenv_safe("TMPDIR", NULL, 0);
-       const char *print_dir = tmpdir;
+       char *tmpdir = getenv_safe("TMPDIR", NULL, 0);
+       char *print_dir = tmpdir;
        if (print_dir == NULL)
                print_dir = "/tmp";
 
        char dir_name[PATH_MAX];
        int rc = snprintf(dir_name, sizeof(dir_name),
                          "%s/tntXXXXXX", print_dir);
-       free((char *)tmpdir);
+       free(tmpdir);
        if (rc < 0 || (size_t)rc >= sizeof(dir_name)) {
                diag_set(SystemError, "failed to generate path to tmp dir");
                goto error;
```